### ISSUE TITLE
When converting class to callable, metaclass `__call__` is not taken into consideration

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7685]
The newly-updated typing spec indicates that when converting a class to a callable, a type checker should take into account whether the class has a custom metaclass with a `__call__` method. Pyright doesn't currently do this.

Here is the conformance test that currently fails.

```python
class Meta1(type):
    def __call__(cls, *args, **kwargs) -> NoReturn:
        raise NotImplementedError("Class not constructable")


class Class5(metaclass=Meta1):
    """Custom metaclass that overrides type.__call__"""

    def __new__(cls, *args, **kwargs) -> Self:
        """This __new__ is ignored for purposes of conversion"""
        return super().__new__(cls)


class Expected5(Protocol):
    def __call__(self) -> NoReturn: ...


r5 = accepts_callable(Class5)
reveal_type(r5)  # ``def () -> NoReturn``
t5: Expected5 = r5  # OK

try:
    assert_type(r5(), NoReturn)
except:
    pass
```

**Comment by erictraut** [comment]
This is included in pyright 1.1.359, which I just published.
